metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarExecutor.java

```diff
@@ -0,0 +1,130 @@
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.metron.common.stellar.shell;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.commons.lang.StringUtils;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.metron.common.configuration.ConfigurationsUtils;
+import org.apache.metron.common.dsl.Context;
+import org.apache.metron.common.dsl.FunctionResolver;
+import org.apache.metron.common.dsl.MapVariableResolver;
+import org.apache.metron.common.dsl.StellarFunctions;
+import org.apache.metron.common.dsl.VariableResolver;
+import org.apache.metron.common.stellar.StellarProcessor;
+import org.apache.metron.common.utils.JSONUtils;
+
+import java.io.ByteArrayInputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.metron.common.configuration.ConfigurationsUtils.readGlobalConfigBytesFromZookeeper;
+
+/**
+ * Executes Stellar expressions and maintains state across multiple invocations.
+ */
+public class StellarExecutor {
+
+  /**
+   * The variables known by Stellar.
+   */
+  private Map<String, Object> variables;
+
+  /**
+   * The function resolver.
+   */
+  private FunctionResolver functionResolver ;
+
+  /**
+   * The context
+   */
+  private Context context;
+
+  public StellarExecutor() throws Exception {
+    this(null);
+  }
+
+  public StellarExecutor(String zookeeperUrl) throws Exception {
+    this.variables = new HashMap<>();
+    this.functionResolver = new StellarFunctions().FUNCTION_RESOLVER();
+    this.context = getContext(zookeeperUrl);
+  }
+
+  /**
+   * Creates a Context initialized with configuration stored in Zookeeper.
+   * @param zookeeperUrl The Zookeeper URL.
+   */
+  private Context getContext(String zookeeperUrl) throws Exception {
+    Context context = Context.EMPTY_CONTEXT();
+
+    // load global configuration from zookeeper
+    if(StringUtils.isNotBlank(zookeeperUrl)) {
+      try (CuratorFramework client = ConfigurationsUtils.getClient(zookeeperUrl)) {
+        client.start();
+
+        Map<String, Object> global = JSONUtils.INSTANCE.load(
+                new ByteArrayInputStream(readGlobalConfigBytesFromZookeeper(client)),
+                new TypeReference<Map<String, Object>>() {
+                });
+
+        context = new Context.Builder()
+                .with(Context.Capabilities.GLOBAL_CONFIG, () -> global)
```

Comment:
Can we add the `Context.Capabilities.ZOOKEEPER_CLIENT` as well?  Looks like we have it right there on line 81.